### ISSUE TITLE
Cannot change encoding in csv export

### ISSUE THREAD

**Issue by moguzalp** [angular-ui/ui-grid#3771]
It seems that related issue is discussed [here](https://github.com/angular-ui/ng-grid/issues/2646#issuecomment-72191022).

I cannot change the encoding other then `UTF-8`, when I toggle `exporterOlderExcelCompatibility` I can see result in notepad++ as utf-8 and utf-8 without BOM. But I cannot set encoding to windows "windows-1254" or "ascii" or iso-8859-9

What I am trying to achieve UCS-2 little endian which works properly in client pc

when I look at the source at line 16106:

```
        downloadFile: function (fileName, csvContent, exporterOlderExcelCompatibility) {
          var D = document;
          var a = D.createElement('a');
          var strMimeType = 'application/octet-stream;charset=utf-8';
          var rawFile;
          var ieVersion;
          //....
```

By the way, as [comma](https://github.com/angular-ui/ng-grid/issues/2646#issuecomment-74350617) is decimal in my local (tr-TR) so excel seems to not willing to accept is delimiter automatically, 
(Option for delimiter is)[https://github.com/angular-ui/ng-grid/issues/2646#issuecomment-71741657]  "exporterCsvColumnSeparator"  that I found source code which is missing in tutorial documentation

**Comment by PaulL1** [comment]
The delimiter option is documented in the api documentation, not every option is included in the tutorials: http://ui-grid.info/docs/#/api/ui.grid.exporter.api:GridOptions

There is not currently an option to change the charset.  If you wished to submit a PR for that it may be accepted into 3.1.

**Comment by moguzalp** [comment]
@PaulL1 There is not such option but what I am asking is editing source code which does not work neither.

**Comment by PaulL1** [comment]
OK, if you wanted to edit the source, there are a few places that utf-8 is set, depending on which browser you use.  You'd need to put some breakpoints in to work out which one is the one that matters.

**Comment by moguzalp** [comment]
Yes I did it  for version v3.0.0-rc.21-9be0549  with chrome

at line 16109:
```var strMimeType = 'application/octet-stream;charset=iso-8859-9';```
   
and at line 24470:
```
 $templateCache.put('ui-grid/csvLink',
    "<span class=\"ui-grid-exporter-csv-link-span\"><a href=\"data:text/csv;charset=iso-8859-9,CSV_CONTENT\" download=\"FILE_NAME\">LINK_LABEL</a></span>"
  );```

which has not any affect on result.

**Comment by PaulL1** [comment]
Yes, I'm not sure Excel honours charsets, and I'm not sure whether you need to actually create the file in that charset, not just specify the charset that you'd like it to be in.

**Comment by moguzalp** [comment]
Actually it is not directly related to Excel. Because I am seeing result in Notepad++ in encoding section.
[In this thread](https://datatables.net/forums/discussion/comment/50621/#Comment_50621) for jquery datatables it mentions about similar issue but it seems they handled in flash file `copy_csv_xls.swf` to generated csv file.

**Comment by moguzalp** [comment]
[This comment](http://stackoverflow.com/questions/155097/microsoft-excel-mangles-diacritics-in-csv-files#comment5294204_314801) on Stack Overflow works for me after spending hours:

My client has Office 2007 but with out SP3. Once install SP3, `exporterOlderExcelCompatibility: true` is enough to open with UTF-8 encoding. No need to UTF16-LE

**Comment by UnnamedSong** [comment]
@moguzalp 
Thank you very much.
```
exporterOlderExcelCompatibility: true
```
Works well!!

**Comment by taverngoal** [comment]
Thanks,

```
exporterOlderExcelCompatibility: true
```

Works well

**Comment by webmato** [comment]
When I export CSV with any exporter settings characters like `ěščřwžšýčáýá` in Excel I always got `Ä›ĹˇÄŤĹ™wĹľĹˇĂ˝ÄŤĂˇĂ˝Ăˇ` . 

Now the only solution is not to open this CSV directly by Excel, but import CSV data into new Excel file (here we can specify encoding).  

But this is not satisfying solution - I can't learn my customers how to import data and change encoding of existing file. We need easiest possible way - just open CSV with correct encoding what would we set directly in UI Grid Exporter.

**Comment by bogdan-panteleev-mifort** [comment]
Hi, guys. I have the same issue. The problem is that after exporting as CSV sign '£' with 
`exporterOlderExcelCompatibility: true` it works fine for windows. But if I export it in MAC i have wrong sing 'BJ' instead of '£'
